org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java

```diff
@@ -0,0 +1,174 @@
+     */
+    public ArtifactServerUtil(final String mavenCentralBaseUrl, final int connectTimeoutInMilliseconds,
+            final ILFLog log) {
+        this.mavenCentralBaseUrl = mavenCentralBaseUrl;
+        this.connectTimeoutInMilliseconds = connectTimeoutInMilliseconds;
+        this.cachedCheckAccess = checkAccess(log);
+    }
+
+    /**
+     * Checks if the artifact server is accessible with the base URL configured in the constructor of this class.
+     * 
+     * @param log a logger
+     * @return true if the artifact server is accessible under the configured base URL, false otherwise
+     */
+    private boolean checkAccess(final ILFLog log) {
+        // we check the presence of
+        // 'org.apache.maven.plugins:maven-plugin-plugin:jar:3.6.0'
+        final String groupId = "org.apache.maven.plugins";
+        final String artifactId = "maven-plugin-plugin";
+        final String version = "3.6.0";
+        final String urlString = getArtifactUrlString(groupId, artifactId, version);
+        log.info("Checking access to artifact server using URL " + urlString);
+        try {
+            final URL url = new URL(urlString);
+            final URLConnection urlConnection = url.openConnection();
+            urlConnection.setConnectTimeout(connectTimeoutInMilliseconds);
+            InputStream is = urlConnection.getInputStream();
+            is.close();
+            return true;
+        } catch (IOException e) {
+            log.warn("Cannot access artifact server at: " + urlString);
+            log.warn("Result is cached, POM resolution is disabled for this run!");
+            return false;
+        }
+    }
+
+    /**
+     * @return the cachedCheckAccess
+     */
+    public final boolean isCachedCheckAccess() {
+        return cachedCheckAccess;
+    }
+
+    /**
+     * @return the nexusCentralBaseUrl
+     */
+    private final String getNexusCentralBaseUrl() {
+        return mavenCentralBaseUrl;
+    }
+
+    /**
+     * Adds licenses to an archive from information found in a Maven POM file.
+     * 
+     * @param inputStream      input source of the POM file
+     * @param archive          the archive to add the licenses to
+     * @param filePath         path of the POM file (for information only)
+     * @param licenseStoreData
+     * @param log              the logger
+     * @return true if one or more licenses have been added, false otherwise
+     */
+    public boolean addLicensesFromPom(final InputStream inputStream, final Archive archive, final String filePath,
+                                      final LicenseStoreData licenseStoreData, final ILFLog log) {
+        try {
+            log.debug("Checking POM file: " + filePath);
+            final MavenXpp3Reader xpp3Reader = new MavenXpp3Reader();
+            final Model model = xpp3Reader.read(inputStream);
+
+            final List<org.apache.maven.model.License> licenses = model.getLicenses();
+            boolean licenseFound = false;
```

Comment:
Move this code to a separate method that return a boolean so to make the addLicensesFromProm method less complex.